dialects/mem/mem.cpp

```diff
@@ -22,17 +23,33 @@ using namespace thorin;
 extern "C" THORIN_EXPORT DialectInfo thorin_get_dialect_info() {
     return {"mem",
             [](PipelineBuilder& builder) {
-        builder.extend_opt_phase([](PassMan& man) {
-            auto br = man.add<BetaRed>();
-            auto er = man.add<EtaRed>();
-            auto ee = man.add<EtaExp>(er);
-            man.add<mem::SSAConstr>(ee);
-            man.add<mem::CopyProp>(br, ee);
-        });
-        builder.extend_codegen_prep_phase([](PassMan& man) {
-            man.add<mem::RememElim>();
-            man.add<mem::Alloc2Malloc>();
-        });
+                builder.extend_opt_phase([](PassMan& man) {
+                    auto br = man.add<BetaRed>();
+                    auto er = man.add<EtaRed>();
+                    auto ee = man.add<EtaExp>(er);
+                    man.add<mem::SSAConstr>(ee);
+                    man.add<mem::CopyProp>(br, ee);
+                });
+                builder.extend_codegen_prep_phase([](PassMan& man) {
+                    man.add<mem::RememElim>();
+                    man.add<mem::Alloc2Malloc>();
+                });
             },
             nullptr, [](Normalizers& normalizers) { mem::register_normalizers(normalizers); }};
 }
+
+namespace thorin::mem {
+
+const Def* op_malloc(const Def* type, const Def* mem, const Def* dbg) {
```

Comment:
I understand the urge for non-inline constructors, but moving this here means we have to export this function explicitly and other dialects have to link against the mem dialect plugin if we want them to be able to use this...